### PR TITLE
SOC-2600 | Inline Editor styles for Discussions

### PR DESCRIPTION
**Pull request by kvas-damian**
## Links

* https://wikia-inc.atlassian.net/browse/SOC-2600

## Description

Styles for inline desktop editor for discussions.

## Reviewers
@bkoval @rafalkalinski

### CODE REVIEW

File: front/main/app/components/discussion-inline-editor.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+	/**
+	 * Set right height for editor placeholder when editor gets sticky
+	 *
+	 * @returns {void}
+	 */
+	style: Ember.computed('isSticky', function () {
+		if (this.get('isSticky')) {
+			const editorHeight = this.$('.discussion-inline-editor-floating-container').outerHeight(true);
+
+			return Ember.String.htmlSafe(`height: ${editorHeight}px`);
```
Comment (bkoval) [inline front/main/app/components/discussion-inline-editor.js]: Why do we need to escape it?

File: front/main/app/components/discussion-inline-editor.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+	/**
+	 * @returns {void}
+	 */
+	onScroll() {
+		Ember.run.throttle(
+			this,
+			function () {
+				// we can't change it to
+				// this.set('isSticky', !this.get('isSticky') && this.isStickyBreakpointHeight())
+				// because it is important to fire the set method only when it's necessary, because there is observer
```
Comment (bkoval) [inline front/main/app/components/discussion-inline-editor.js]: I don't such an observer in the code, I know it's a copied comment but maybe we could update it?

File: front/main/app/styles/component/_discussion-inline-editor.scss [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+	}
+
+	&.is-sticky {
+		.discussion-inline-editor-floating-container {
+			border-top: 0;
+			margin-left: ($discussion-page-width - $sticky-editor-width) / 2;
+			position: fixed;
+			top: $site-head-height;
+			width: $sticky-editor-width;
+			z-index: 1;
```
Comment (bkoval) [inline front/main/app/styles/component/_discussion-inline-editor.scss]: Why is that needed? Is editor unable to cover something on the page? If it's needed, then we could define a variable for it (and use one of style-guide values).

File: front/main/app/styles/component/_discussion-inline-editor.scss [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+			border: 0;
+			min-height: 106px;
+			resize: vertical;
+		}
+	}
+
+	&.is-sticky {
+		.discussion-inline-editor-floating-container {
+			border-top: 0;
+			margin-left: ($discussion-page-width - $sticky-editor-width) / 2;
```
Comment (bkoval) [inline front/main/app/styles/component/_discussion-inline-editor.scss]: Can this produce a margin < 0?

File: front/main/app/styles/component/_discussion-inline-editor.scss [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+			position: fixed;
+			top: $site-head-height;
+			width: $sticky-editor-width;
+			z-index: 1;
+		}
+	}
+
+	button {
```
Comment (bkoval) [inline front/main/app/styles/component/_discussion-inline-editor.scss]: What do you think about adding a classname to the submit button and using it in the SCSS? It's possible we'll add some more buttons to the editor in the future and this would prevent accidental inheritance of styles.

File: front/main/app/styles/component/_discussion-inline-editor.scss [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+		background: none;
+		border: 0;
+		font-size: $type-smallest;
+		font-weight: $font-weight-bold;
+		height: 40px;
```
Comment (bkoval) [inline front/main/app/styles/component/_discussion-inline-editor.scss]: Do we need to set a height on it? Maybe line-height and paddings would be enough?

File: front/main/app/styles/component/_discussion-inline-editor.scss [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+		letter-spacing: .25px;
+		margin: 0;
+		padding: 0 1.25rem;
```
Comment (bkoval) [inline front/main/app/styles/component/_discussion-inline-editor.scss]: I think we should use pixel values for paddings :)

File: front/main/app/styles/component/_discussion-open-graph.scss [hunk 8]
```diff
@@ -15,7 +15,8 @@
 
 .discussions {
 	// -- theme --
```
Comment (bkoval) [inline front/main/app/styles/component/_discussion-open-graph.scss]: Let's just remove this ridicoulus thing. If we want to separate theme and layout styles we should split them into two files and here we're mixing them anyway :)

File: front/main/app/templates/components/discussion-inline-editor.hbs [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+			{{textarea
+				autoresize=true
+				class="discussion-inline-editor-textarea"
+				placeholder=(i18n placeholderMessageKey ns='discussion')
+				value=content
+			}}
+		</div>
+		{{#if showsOpenGraphCard}}
+			{{discussion-open-graph
+			close=(action 'removeOpenGraph')
```
Comment (bkoval) [inline front/main/app/templates/components/discussion-inline-editor.hbs]: One indentation more ;>

File: front/main/app/templates/components/discussion-inline-editor.hbs [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+			isLoading=isOpenGraphLoading
+			openGraphData=openGraph
+			}}
+		{{/if}}
+	</div>
+	<div class="discussion-inline-editor-toolbar">
+		<button class="active-element-theme-color" type="submit" {{action 'submit' bubbles=false}} disabled={{submitDisabled}}>{{i18n submitMessageKey ns='discussion'}}</button>
```
Comment (bkoval) [inline front/main/app/templates/components/discussion-inline-editor.hbs]: Did we clear things up about bubbles=false yet? :)

File: front/main/app/templates/components/discussion-inline-editor.hbs [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+	</div>
+
+	{{#if isLoading}}
+		{{loading-spinner
+		active=isLoading
```
Comment (bkoval) [inline front/main/app/templates/components/discussion-inline-editor.hbs]: Indentation

File: front/main/app/components/discussion-inline-editor.js (hunk 1 above)
Comment (kvas-damian) [inline front/main/app/components/discussion-inline-editor.js]: It's not escaping - it's marking string as safe that doesn't need to be escaped later.

File: front/main/app/components/discussion-inline-editor.js (hunk 2 above)
Comment (kvas-damian) [inline front/main/app/components/discussion-inline-editor.js]: I'm not sure I understand you. Can you elaborate more?

File: front/main/app/styles/component/_discussion-inline-editor.scss (hunk 3 above)
Comment (kvas-damian) [inline front/main/app/styles/component/_discussion-inline-editor.scss]: It's required because we have containers with relative position on the same level. I'll add the variable.

File: front/main/app/styles/component/_discussion-inline-editor.scss (hunk 4 above)
Comment (kvas-damian) [inline front/main/app/styles/component/_discussion-inline-editor.scss]: We're expecting negative margin here to center sticky editor which is wider than regular one.

File: front/main/app/styles/component/_discussion-open-graph.scss (hunk 8 above)
Comment (kvas-damian) [inline front/main/app/styles/component/_discussion-open-graph.scss]: I didn't touch this code, let's do it in separate PR to limit this PR only to required changes.

File: front/main/app/templates/components/discussion-inline-editor.hbs (hunk 10 above)
Comment (kvas-damian) [inline front/main/app/templates/components/discussion-inline-editor.hbs]: We have click action bound to entire editor component that opens editor. In order to avoid race condition we need to prevent bubbling on this button.

File: front/main/app/components/discussion-inline-editor.js (hunk 1 above)
Comment (bkoval) [inline front/main/app/components/discussion-inline-editor.js]: 💛

File: front/main/app/styles/component/_discussion-inline-editor.scss (hunk 4 above)
Comment (bkoval) [inline front/main/app/styles/component/_discussion-inline-editor.scss]: 🆗

**Comment by bkoval** [comment]
LGTM 🌴
